src/HoverProvider.ts

```diff
@@ -0,0 +1,43 @@
+import * as vscode from 'vscode';
+import * as yaml from 'yaml';
+
+export class AsyncAPIHoverProvider implements vscode.HoverProvider {
+  provideHover(
+    document: vscode.TextDocument,
+    position: vscode.Position,
+    token: vscode.CancellationToken
+  ): vscode.ProviderResult<vscode.Hover> {
+    const lineText = document.lineAt(position.line).text;
+    const wordRange = document.getWordRangeAtPosition(position, /[\w.$-]+/);
+    const hoveredWord = document.getText(wordRange);
+
+    // Parse the document to get structured context (optional for later)
+    // const parsedYaml = yaml.parse(document.getText());
+
+    // Show hover based on key presence in line
+    const hoverTexts: { [key: string]: string } = {
```

Comment:
good for start testing the water of a HoverProvider

Have a look to this library: https://stoplightio.github.io/yaml/
It let's you, from a file prosition, know the json path that location is in
with the location of this json path (ex `#/components/schemas/UserSchema`) you can build some usefull information to display about that node, maybe a class diagram for the `UserSchema`